rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -163,84 +162,23 @@ std::string make_error_str(
 
   return error.dump();
 }
-} // anonymous namespace
 
 //==============================================================================
-void FleetUpdateHandle::Implementation::bid_notice_cb(
-  const BidNoticeMsg& bid_notice,
-  rmf_task_ros2::bidding::AsyncBidder::Respond respond)
+std::shared_ptr<rmf_task::Request> FleetUpdateHandle::Implementation::convert(
+  const std::string& task_id,
+  const nlohmann::json& request_msg,
+  std::vector<std::string>* errors) const
 {
-  const auto& task_id = bid_notice.task_id;
-  if (task_managers.empty())
-  {
-    RCLCPP_INFO(
-      node->get_logger(),
-      "Fleet [%s] does not have any robots to accept task [%s]. Use "
-      "FleetUpdateHadndle::add_robot(~) to add robots to this fleet. ",
-      name.c_str(), task_id.c_str());
-    return;
-  }
-
-  if (task_id.empty())
-  {
-    RCLCPP_WARN(
-      node->get_logger(),
-      "Received BidNotice for a task with empty task_id. Request will be "
-      "ignored.");
-    return;
-  }
-
-  // TODO remove this block when we support task revival
-  if (bid_notice_assignments.find(task_id) != bid_notice_assignments.end())
-    return;
-
-  if (!task_planner)
-  {
-    RCLCPP_WARN(
-      node->get_logger(),
-      "Fleet [%s] is not configured with parameters for task planning."
-      "Use FleetUpdateHandle::set_task_planner_params(~) to set the "
-      "parameters required.", name.c_str());
-
-    return;
-  }
-
-  const auto request_msg = nlohmann::json::parse(bid_notice.request);
-  static const auto request_validator =
-    nlohmann::json_schema::json_validator(rmf_api_msgs::schemas::task_request);
-
-  try
-  {
-    request_validator.validate(request_msg);
-  }
-  catch (const std::exception& e)
-  {
-    RCLCPP_ERROR(
-      node->get_logger(),
-      "Received a request with an invalid format. Error: %s\nRequest:\n%s",
-      e.what(),
-      request_msg.dump(2, ' ').c_str());
-
-    return respond(
-      {
-        std::nullopt,
-        {make_error_str(5, "Invalid request format", e.what())}
-      });
-  }
-
   const auto& category = request_msg["category"].get<std::string>();
 
   const auto task_deser_it = deserialization.task->handlers.find(category);
   if (task_deser_it == deserialization.task->handlers.end())
   {
-    return respond(
-      {
-        std::nullopt,
-        {make_error_str(
-          4, "Unsupported type",
-          "Fleet [" + name + "] does not support task category ["
-          + category + "]")}
-      });
+    errors->push_back(make_error_str(
```

Comment:
Are we guaranteed to always be provided a non-null `errors`? If so, why not pass in `std::vector<std::string>&` instead? If not, we need to check `if (errors)` here and also in all the places that we use `errors` below.

Reply:
I should be checking `if (errors)` which I missed out. Not using `std::vector<std::string>&` to retain the `const` correctness of the function

Reply:
I don't see how the choice between a pointer and a reference would impact const correctness. The only effective difference between the two is that references are not (supposed to be) nullable. If we always expect the caller of the function to provide a valid reference to a `std::vector<std::string>` object, then we should give the function a mutable reference argument `std::vector<std::string>&`.

Reply:
Whoops I was totally wrong here. Must have gotten confused with an unrelated error.